File: materials/forecast-java/src/main/java/com/forecast/controllers/weather/CurrentWeatherController.java
package com.forecast.controllers.weather;

import com.forecast.clients.WeatherDataClient;
import com.forecast.models.weather.CurrentWeather;
import java.math.BigDecimal;

public class CurrentWeatherController {
    private final WeatherDataClient client;

    public CurrentWeatherController(WeatherDataClient client) {
        this.client = client;
    }

    public CurrentWeather getCurrentWeather(BigDecimal lat, BigDecimal lon) {
        try {
            BigDecimal temperature = client.locationCurrentTemperature(lat, lon);
            return new CurrentWeather(temperature);
        } catch (Exception e) {
            return null;
        }
    }
}

File: materials/forecast-csharp/Controllers/CurrentWeatherController.cs
using Forecast.Clients;
using Forecast.Models.Weather;

namespace Forecast.Controllers;

class CurrentWeatherController(IWeatherDataClient client)
{
    private readonly IWeatherDataClient client = client;

    public async Task<CurrentWeather> GetCurrentWeather(decimal latitude, decimal longitude)
    {
        var temperature = await client.GetCurrentTemperatureAtLocation(latitude, longitude);
        return new(temperature);
    }
}
